src/libraries/System.Runtime.Extensions/tests/System/Math.cs

```diff
@@ -796,6 +796,24 @@ public static void Max_Decimal()
         public static void Max_Double_NotNetFramework(double x, double y, double expectedResult)
         {
             AssertExtensions.Equal(expectedResult, Math.Max(x, y), 0.0);
+
+            if (double.IsNaN(x))
+            {
+                ulong bits = BitConverter.DoubleToUInt64Bits(x);
+                bits ^= BitConverter.DoubleToUInt64Bits(-0.0);
+                x = BitConverter.UInt64BitsToDouble(bits);
```

Comment:
What does xor'ing a NaN with -0 do?

Reply:
Floating-point is one's complement so it toggles the sign, but in a way that works for NaN

Reply:
Added a comment:
```csharp
                // Toggle the sign of the NaN to validate both +NaN and -NaN behave the same.
                // Negate should work as well but the JIT may constant fold or do other tricks
                // and normalize to a single NaN form so we do bitwise tricks to ensure we test
                // the right thing.
```